packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1522,6 +1538,17 @@ export class ContainerRuntime
 		}
 	}
 
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IRuntime.getEntryPoint}
+	 */
+	public async getEntryPoint(): Promise<FluidObject | undefined> {
+		if (this._entryPoint === undefined && this.initializeEntryPoint !== undefined) {
+			this._entryPoint = await this.initializeEntryPoint?.(this);
+		}
+		return this._entryPoint;
+	}
+	private _entryPoint: FluidObject | undefined;
```

Comment:
I think we can simplify this in two ways. get rid of one member, and have the only existing be immutable.
`private readonly entrypoint: LazyPromise<FluidObject | undefined`
then in the constructor
```
this.entrypoint = new LazyPromise(()=>initializeEntryPoint?.(this));
```

Reply:
this getter than just returns `this.entryPoint`

Reply:
Need to get used to LazyPromise 😄 . Done!